vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/utils/DateUtils.kt

```diff
@@ -15,27 +16,67 @@ import io.mosip.vercred.vcverifier.exception.ValidationException
 import org.json.JSONObject
 import java.text.SimpleDateFormat
 import java.util.Calendar
+import java.util.Date
 import java.util.Locale
 import java.util.TimeZone
 
-class DateUtils {
+object DateUtils {
+
+    private val dateFormats = listOf(
+        ("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"),
+        ("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    )
+
+    private const val UTC = "UTC"
+
     fun isValidDate(dateValue: String): Boolean {
         return DATE_REGEX.matches(dateValue)
     }
 
     fun isDatePassedCurrentDate(inputDateString: String): Boolean {
         return try {
-            val format = SimpleDateFormat(COMMON_DATE_FORMAT, Locale.getDefault()).apply {
-                timeZone = TimeZone.getTimeZone(UTC)
+            val inputDate: Date? = parseDate(inputDateString)
+            if (inputDate == null) {
+                Log.e("VC-VERIFIER", "Given date is not available in supported date formats")
+                return false
             }
-            val inputDate = format.parse(inputDateString)
+
             val currentDate = Calendar.getInstance(TimeZone.getTimeZone(UTC)).time
-            inputDate.before(currentDate)
+            inputDate!!.before(currentDate)
         } catch (e: Exception) {
+            Log.e("VC-VERIFIER", "Error while comparing dates ${e.message}")
             false
         }
     }
 
+    fun isDate1GreaterThanDate2(date1: String, date2: String): Boolean {
```

Comment:
what is this function about?

Reply:
we have a check if validUntil > validFrom in mso mdoc for that have extracted this function

Reply:
It's looking more generic
as a consumer I can pass date1 as validFrom. So better to have name specific

Reply:
Tried Avoided specificNames of VC field names here considering this is a utility class specific for checking dates related functions. Its mentioned in the MdocValidator class on what check is performed as mentioned below, should we mention reg validUntilAndValidFromDates here?

```val isValidUntilGreaterThanValidFrom = DateUtils.isDate1GreaterThanDate2(
                validUntil.toString(),
                validFrom.toString()
            )

Reply:
As per discussion, To avoid these  naming confusions comparison of dates is not handled in validator using DateUtils parse method to get the date instances from stringDate